openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,822 @@
+        # Set _prepared to False to ensure parameter subset
+        # used during construction routine
+        res_cov_range.file2res._prepared = False
+
+        return res_cov_range
+
+    def sample_resonance_parameters(self, n_samples):
+        """Sample resonance parameters based on the covariances provided
+        within an ENDF evaluation.
+
+        Parameters
+        ----------
+        n_samples : int
+            The number of samples to produce
+
+        Returns
+        -------
+        samples : list of openmc.data.ResonanceCovarianceRange objects
+            List of samples size `n_samples`
+
+        """
+        warn_str = 'Sampling routine does not guarantee positive values for '\
+                   'parameters. This can lead to undefined behavior in the '\
+                   'reconstruction routine.'
+        warnings.warn(warn_str)
+        parameters = self.parameters
+        cov = self.covariance
+        # Copy ResonanceRange object
+        res_range = copy.copy(self.file2res)
+        # Set _prepared to False to ensure sampled parameters are
+        # used during construction routine
+        res_range._prepared = False
+
+        nparams, params = parameters.shape
+        # Symmetrizing covariance matrix
+        cov = cov + cov.T - np.diag(cov.diagonal())
+        covsize = cov.shape[0]
+        formalism = self.formalism
+        mpar = self.mpar
+        samples = []
+
+        # Handling MLBW sampling
+        if formalism == 'mlbw' or formalism == 'slbw':
+            if mpar == 3:
+                param_list = ['energy', 'neutronWidth', 'captureWidth']
+                mean_array = pd.DataFrame.as_matrix(parameters[param_list])
+                spin = pd.DataFrame.as_matrix(parameters['J'])
+                l_value = pd.DataFrame.as_matrix(parameters['L'])
+                gf = pd.DataFrame.as_matrix(parameters['fissionWidth'])
+                gx = pd.DataFrame.as_matrix(parameters['competitiveWidth'])
+                mean = mean_array.flatten()
+                par_samples = np.random.multivariate_normal(mean, cov,
+                                                            size=n_samples)
+                for sample in par_samples:
+                    energy = sample[0::3]
+                    gn = sample[1::3]
+                    gg = sample[2::3]
+                    gt = gn + gg + gf
+                    records = []
+                    for j, E in enumerate(energy):
+                        records.append([energy[j], l_value[j], spin[j], gt[j],
+                                        gn[j], gg[j], gf[j], gx[j]])
+                    columns = ['energy', 'L', 'J', 'totalWidth', 'neutronWidth',
+                               'captureWidth', 'fissionWidth', 'competitiveWidth']
+                    sample_params = pd.DataFrame.from_records(records,
+                                                              columns=columns)
+                    res_range.parameters = sample_params
```

Comment:
Related to the previous comment, this line overwrites the parameters on each iteration of the for loop. (also, I don't see a similar line updating the parameters for the mpar=4/5 and the RM cases)